src/app/wizards/dataModel/data-model-main/data-model-main.component.ts

```diff
@@ -114,12 +116,16 @@ export class DataModelMainComponent implements OnInit {
       .addToFolder(this.parentFolderId, resource, queryStringParams)
       .pipe(
         catchError(error => {
+          this.disabledSave =  false;
           this.messageHandler.showError('There was a problem saving the Data Model.', error);
           return EMPTY;
         }))
       .subscribe((response: DataModelDetailResponse) => {
         this.messageHandler.showSuccess('Data Model saved successfully.');
+        this.disabledSave =  false;
         this.stateHandler.Go('datamodel', { id: response.body.id }, { reload: true, location: true });
+      }, error => {
+        this.disabledSave =  false;
```

Comment:
There is a shorter way of setting this flag back to `false` without having to duplicate it. There is an `rxjs` operator called `finalize()` which, if included in the `pipe()`, will always be called regardless of whether it was successful or not. Including something like the following would then you only need to set this flag once:

```ts
.addToFolder()
.pipe(
  catchError(error => {
    // Error case
  }),
  finalize(() => this.disabledSave = false)  // Always called at the end
)
.subscribe((response) => {
  // Success case
});
```